sdks/python/apache_beam/transforms/util.py

```diff
@@ -725,7 +725,7 @@ class Reshuffle(PTransform):
   """
 
   # We use 32-bit integer as the default number of buckets.
-  _DEFAULT_NUM_BUCKETS = 1 << 32
+  _DEFAULT_NUM_BUCKETS = (1 << 32) - 1
```

Comment:
https://github.com/python/cpython/blob/main/Lib/random.py

This effectively becomes `randint(0, (1 << 32) - 1)`

randint -> randrange(a, b+1) so we have `randrange(0, (1 << 32))`

For this case randrange(0, n) becomes _randbelow(n) so `_randbelow(1<<32)`

_randbelow(n) becomes getrandbits(n.bit_length()) so `_randbelow(33)`

Seems there is still an off by one error?

Reply:
From a chat discussion, I think either of these are equivalent: `randint(0, (1 << 32) - 2)` or `randrange(0, (1 << 32)-1)`

`randint(0, (1 << 32) - 2)` -> `randrange(0, (1 << 32)-1)` -> `_randbelow((1<<32)-1)` -> `getrandbits(32)`

Reply:
Actually, there is an off by one error in the python: https://bugs.python.org/issue37000

So even though it calls `getrandbits(33)` it rejects the result if it is more than 32 bits and retries. This change is correct.